desk/src/components/modals/ShortcutsModal.vue

```diff
@@ -0,0 +1,128 @@
+<template>
+  <Dialog
+    v-model="open"
+    :options="{ title: 'Keyboard Shortcuts', size: '4xl' }"
+  >
+    <template #body-content>
+      <div class="w-full grid grid-cols-2 gap-10 py-1 shortcutsModal">
+        <div
+          v-for="group in shortcutGroups"
+          :key="group.title"
+          class="pb-4"
+          :class="!group.hideBorder && 'border-b border-outline-gray-2'"
+        >
+          <h2 class="text-lg font-semibold text-ink-gray-9 mb-4">
+            {{ group.title }}
+          </h2>
+          <ul class="space-y-2">
+            <li
+              v-for="(shortcut, index) in group.shortcuts"
+              :key="index"
+              class="flex items-start justify-between gap-4"
+            >
+              <div class="text-ink-gray-7 text-base flex-1">
+                {{ shortcut.description }}
+              </div>
+              <div class="flex space-x-1 gap-1 justify-end">
+                <span
+                  v-for="(key, kIndex) in shortcut.keys"
+                  :key="kIndex"
+                  class="bg-surface-gray-2 border border-outline-gray-2 text-xs rounded-sm text-ink-gray-8 shadow-sm min-w-5 h-5 flex items-center justify-center"
+                  :class="[
+                    ![metaIcon, shiftKey].includes(key) && 'w-5 ',
+                    key === shiftKey && '!px-2',
+                  ]"
+                >
+                  {{ key }}
+                </span>
+              </div>
+            </li>
+          </ul>
+        </div>
+      </div>
+    </template>
+  </Dialog>
+</template>
+
+<script setup lang="ts">
+import { useDevice } from "@/composables/device";
+import { useShortcut } from "@/composables/shortcuts";
+import { isCustomerPortal } from "@/utils";
+import { Dialog } from "frappe-ui";
+import { computed, onMounted } from "vue";
+
+const props = defineProps({
+  modelValue: Boolean,
+});
+const emit = defineEmits(["update:modelValue"]);
+const open = defineModel<boolean>();
+
+const { metaIcon } = useDevice();
+
+// const shiftKey = "⇧";
+const shiftKey = "Shift";
+
+interface Shortcut {
+  keys: string[];
+  description: string;
+}
+
+interface ShortcutGroup {
+  title: string;
+  shortcuts: Shortcut[];
+  hideBorder?: boolean;
+}
+
+const shortcutGroups = computed<ShortcutGroup[]>(() => [
```

Comment:
One thing that would have made this really cool is if you somehow pulled active shortcuts for the current view.

---

Ideally, your `useShortcut` composable should take "Shortcut Description" and "Group Title" and should have an internal system to list active shortcuts. You can maintain a computed value called `activeShortcuts`, which updates on shortcut attach and destroy.

Modals like this will just use a reactive variable such as `activeShortcuts` to display it.

Once this is done, we'll have a single source of truth and won’t have to maintain a separate map for shortcuts. This can then be part of frappe-ui, as it will be generic enough.